cmd/sonobuoy/app/images.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/image"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var imagesflags imagesFlags
+
+type imagesFlags struct {
+	e2eRegistryConfig string
+	kubeconfig        Kubeconfig
+
+	imagesflags *pflag.FlagSet
+}
+
+func ImagesFlagSet(cfg *imagesFlags) *pflag.FlagSet {
+	flagset := pflag.NewFlagSet("images", pflag.ExitOnError)
+	AddKubeconfigFlag(&cfg.kubeconfig, flagset)
+	cfg.imagesflags = flagset
+	return flagset
+}
+
+func NewCmdImages() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:       "images",
+		Short:     "Manage images used in a plugin. Supported plugins are: 'e2e'",
+		Run:       getImages,
+		ValidArgs: []string{"e2e"},
+		Args:      matchAll(cobra.MinimumNArgs(1), cobra.OnlyValidArgs),
```

Comment:
Would it be better to make `--plugin` a flag that defaults to `e2e`?

Then the user could provide 
```
sonobuoy images
sonobuoy images --plugin e2e // Same as the above
sonobuoy images --plugin foo // one day if another plugin supports this
```

The description of `--plugin` and which are supported values would then go onto the flag.

I tried it out and the cobra generated help is pretty confusing because of this. It doesnt show the `short` description during the error so it doesnt tell you at all what you really need to put for the arg (which to a user is odd because only one value is allowed anyways)

![image](https://user-images.githubusercontent.com/10273533/54152898-505bca80-440c-11e9-811f-60475a5837d0.png)

Reply:
Ok we discussed this previously: https://github.com/heptio/sonobuoy/issues/604#issuecomment-467031910

I can use `--plugin` will fit better with `download`, `upload`, etc.

Reply:
:-D Well what does that other guy know!

I was running an idea there but def didnt think of the implications with cobra; that works out worse than I would have thought.

The other way the original logic could work OK with cobra is to make it a subcommand, not an argument; that way cobra would suggest it as a subcommand, give additional help, etc. I'm surprised cobra doesn't make the valid args more clear tbh. It would have looked the same but played nicer with all the doc gen stuff.

Reply:
Nah I think this works in the end.